### PR TITLE
Host metadata API

### PR DESCRIPTION
**Pull request by yuuki**
# What?

- Implement APIs for Host metadata https://mackerel.io/ja/api-docs/entry/metadata.

# How to check?

https://gist.github.com/yuuki/acd9c06a591f13feafbeaad4c7e3c673

### CODE REVIEW

File: host_metadata.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+// https://mackerel.io/ja/api-docs/entry/metadata
+
+// HostMetaDataResp represents response for host metadata.
+type HostMetaDataResp struct {
+	HostMetaData HostMetaData
+	LastModified time.Time
+}
+
+// HostMetaData represents host metadata body.
+type HostMetaData map[string]interface{}
```
Comment (astj) [inline host_metadata.go]: Host metadata accepts ANY *JSON (object, array, string, number, boolean, null)* as value, so this interface is not enough.
ref: https://mackerel.io/api-docs/entry/metadata

File: host_metadata.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		return nil, err
+	}
+	namespaces := make([]string, 0, len(data.MetaDatas))
+	for _, metadata := range data.MetaDatas {
+		namespaces = append(namespaces, metadata.NameSpace)
+	}
+	return namespaces, nil
+}
+
+// CreateHostMetaData create host metadata.
```
Comment (astj) [inline host_metadata.go]: [suggestion]
How about just using `PutHostMetaData` ?

**Comment by astj** [comment]
@yuuki Thank you for fixing.
I wrote a suggestion. What do you think?

File: host_metadata.go (hunk 2 above)
Comment (yuuki) [reply host_metadata.go]: That sounds good. Why did I do so ...

**Comment by astj** [comment]
Thank You! LGTM

**Pull request merged**